Artifacts/windows-desired-state-configuration-az-automation/DscMetaConfigs.ps1

```diff
@@ -0,0 +1,108 @@
+# The DSC configuration that will generate metaconfigurations
+# Courtesy of https://docs.microsoft.com/en-us/azure/automation/automation-dsc-onboarding#secure-registration
+[DscLocalConfigurationManager()]
+Configuration DscMetaConfigs
+{
+
+    param
+    (
+        [Parameter(Mandatory=$True)]
+        [String]$RegistrationUrl,
+
+        [Parameter(Mandatory=$True)]
+        [String]$RegistrationKey,
+
+        [Parameter(Mandatory=$True)]
+        [String[]]$ComputerName,
+
+        [Int]$RefreshFrequencyMins = 30,
+
+        [Int]$ConfigurationModeFrequencyMins = 15,
+
+        [String]$ConfigurationMode = "ApplyAndMonitor",
+
+        [String]$NodeConfigurationName,
+
+        [Boolean]$RebootNodeIfNeeded= $False,
+
+        [String]$ActionAfterReboot = "ContinueConfiguration",
+
+        [Boolean]$AllowModuleOverwrite = $False,
+
+        [Boolean]$ReportOnly
+    )
+
+    if(!$NodeConfigurationName -or $NodeConfigurationName -eq "")
+    {
+        $ConfigurationNames = $null
+    }
+    else
+    {
+        $ConfigurationNames = @($NodeConfigurationName)
+    }
+
+    if($ReportOnly)
+    {
+    $RefreshMode = "PUSH"
+    }
+    else
+    {
+    $RefreshMode = "PULL"
+    }
+
+    Node $ComputerName
+    {
+
+        Settings
+        {
+            RefreshFrequencyMins = $RefreshFrequencyMins
+            RefreshMode = $RefreshMode
+            ConfigurationMode = $ConfigurationMode
+            AllowModuleOverwrite = $AllowModuleOverwrite
+            RebootNodeIfNeeded = $RebootNodeIfNeeded
+            ActionAfterReboot = $ActionAfterReboot
+            ConfigurationModeFrequencyMins = $ConfigurationModeFrequencyMins
+        }
+
+        if(!$ReportOnly)
+        {
+        ConfigurationRepositoryWeb AzureAutomationDSC
+            {
+                ServerUrl = $RegistrationUrl
+                RegistrationKey = $RegistrationKey
+                ConfigurationNames = $ConfigurationNames
+            }
+
+            ResourceRepositoryWeb AzureAutomationDSC
+            {
+            ServerUrl = $RegistrationUrl
+            RegistrationKey = $RegistrationKey
+            }
+        }
+
+        ReportServerWeb AzureAutomationDSC
+        {
+            ServerUrl = $RegistrationUrl
+            RegistrationKey = $RegistrationKey
+        }
+    }
+}
+    # Create the metaconfigurations
+$Params = @{
+    RegistrationUrl = "$($RegistrationUrl)";
+    RegistrationKey = "$($RegistrationKey)";
+    ComputerName = @($Machine);
+    NodeConfigurationName = "$($ConfigName)";
+    RefreshFrequencyMins = $RefreshMinutes;
+    ConfigurationModeFrequencyMins = $ConfigMinutes;
+    RebootNodeIfNeeded = $Reboot;
```

Comment:
I am not sure if this would work. You need to explicitly configure your artifact in a specific way for it support reboots. Refer sysprep / deprovision artifact.

Reply:
This particular script is only for bootstrapping the registration of the machine with the DSC server.  The reboot parameter is being passed to the DSC server to instruct it whether or not to reboot the machine if a configuration requires it.  It is not requiring a reboot on behalf of the artifact.